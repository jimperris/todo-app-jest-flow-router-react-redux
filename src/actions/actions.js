//@flow
import type {TodoFilterActions} from '../types/basic'
var num: number = 0
export const addTodo = (title: string): TodoFilterActions => {
  return {
    type: 'ADD_TODO',
    id: num++,
    title: title
  }
}
export const removeTodo = (id: number): TodoFilterActions => {
  return {
    type: 'REMOVE_TODO',
    id: id
  }
}

export const setVisibilityFilter = (filter: string) => {
  return {
    type: 'SET_VISIBILITY_FILTER',
    filter: filter
  }
}

export const toggleTodo = (id: number) => {
  return {
    type: 'TOGGLE_TODO',
    id: id
  }
}

export const VisibilityFilters = {
  SHOW_ALL: 'SHOW_ALL',
  SHOW_COMPLETED: 'SHOW_COMPLETED',
  SHOW_IN_PROGRESS: 'SHOW_IN_PROGRESS'
}
