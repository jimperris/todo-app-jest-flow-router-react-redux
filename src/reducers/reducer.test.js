import reducer from './root'
import {createStore} from 'redux'
import * as actions from '../actions/actions'

const store = createStore(reducer)
beforeEach(() => {
  store.dispatch({type: 'RESET'})
})
it('todo reducer should initiate with correct state', () => {
  expect(store.getState()).toMatchSnapshot()
  expect(store.getState()).toEqual({ todos: [{title: 'add', id: 9999, complete: false}], filter: 'SHOW_ALL'})
})

it('should change visibilityFilter string when called', () => {
  store.dispatch(actions.setVisibilityFilter(actions.VisibilityFilters.SHOW_IN_PROGRESS))
  expect(store.getState().filter).toEqual('SHOW_IN_PROGRESS')
})

it('should create new todo object when called', () => {
  let expectedState = { todos: [{title: 'add', id: 9999, complete: false}, {title: 'darude', id: 0, complete: false}], filter: 'SHOW_ALL' }
  store.dispatch(actions.addTodo('darude'))
  expect(store.getState()).toEqual(expectedState)
})

it('should delete todo when called', () => {
  store.dispatch(actions.removeTodo('23'))
  expect(store.getState()).toEqual({todos: [], filter: 'SHOW_ALL'})
})

it('should toggle a todos completion when called', () => {
  let expectedState = [{title: 'add', id: 9999, complete: true}]
  store.dispatch(actions.toggleTodo(9999))
  expect(store.getState().todos).toEqual(expectedState)
})
