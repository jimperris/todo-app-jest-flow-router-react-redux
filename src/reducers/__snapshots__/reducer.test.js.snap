// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo reducer should initiate with correct state 1`] = `
Object {
  "filter": "SHOW_ALL",
  "todos": Array [
    Object {
      "complete": false,
      "id": 9999,
      "name": "add",
    },
  ],
}
`;
